client/my-sites/checkout/src/payment-methods/credit-card/credit-card-pay-button.tsx

```diff
@@ -201,33 +251,68 @@ function isCreditCardFormValid(
 			let isValid = true;
 
 			const rawState = selectors.getFields( store.getState() );
-			const cardholderName = rawState.cardholderName;
-			const numberWithoutSpaces = {
-				value: rawState?.number?.value?.replace( /\s+/g, '' ),
-			}; // the validator package we're using requires this
-			const paymentDetailsData = {
-				...rawState,
-				country: rawState.countryCode,
-				name: cardholderName,
-				number: numberWithoutSpaces,
-			};
-			const validationResults = validatePaymentDetails(
-				Object.entries( paymentDetailsData ).reduce< Record< string, string > >(
-					( accum, [ key, managedValue ] ) => {
-						accum[ key ] = managedValue?.value;
-						return accum;
-					},
-					{}
-				),
-				'ebanx'
-			);
-			Object.entries( validationResults.errors ).map( ( [ key, errors ] ) => {
-				errors.map( ( error ) => {
-					isValid = false;
-					store.dispatch( actions.setFieldError( key, error ) );
+
+			// Check if we're using VGS (determined by feature flag at runtime)
+			// Simple heuristic: if card number field is empty, we're likely using VGS
+			const isUsingVgs = ! rawState?.number?.value;
+
+			if ( isUsingVgs ) {
+				// For VGS mode, we validate billing fields only here
+				// VGS card fields validation is handled separately via useVgsFormValidation hook
+				// which checks the VGS form state directly
+				const requiredFields = [
+					'cardholderName',
+					'state',
+					'city',
+					'postal-code',
+					'address-1',
+					'street-number',
+					'phone-number',
+					'document',
+				];
+
+				requiredFields.forEach( ( fieldName ) => {
+					const fieldValue = rawState[ fieldName ]?.value;
+					if ( ! fieldValue || fieldValue.trim() === '' ) {
+						isValid = false;
+						store.dispatch( actions.setFieldError( fieldName, __( 'This field is required' ) ) );
+					}
+				} );
+
+				debug( 'ebanx validation (VGS mode) - cardholder name and contact details', {
+					isValid,
 				} );
-			} );
-			debug( 'ebanx card details validation results: ', validationResults );
+			} else {
```

Comment:
I also think that we could avoid this `else` by returning `isValid` in the above block.